startReactCommerce.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+git checkout main
+git pull
+git checkout -
+rm -rf node_modules
+nvm use
+npm install -g pnpm@8
+npm install -g yarn
+cp .env.local.example .env.local
+
+## Verdaccio setup
+read -p "Do you have a verdaccio account? (y/n)? [y]: " userInput
+
+# Remove leading/trailing whitespace and convert to lowercase
+userInput=$(echo "${userInput}" | tr '[:upper:]' '[:lower:]' | xargs)
+
+# Check the user's input
+while [[ ! "${userInput}" =~ ^(y|n)$ ]]; do
+    if [[ -z "${userInput}" ]]; then
+        userInput="y"  # Set default value to "yes" if no input is provided
+        break
+    fi
+    echo "Invalid input. Please enter 'y' or 'n'."
+    read -p "Do you have a verdaccio account? (y/n)? [y]: " userInput
+    userInput=$(echo "${userInput}" | tr '[:upper:]' '[:lower:]' | xargs)
+done
+
+# Continue based on the user's input
+if [[ "${userInput}" == "y" ]]; then
+    # Login to exisiting user:
+    pnpm login --registry=https://verdaccio.ubreakit.com    
+else
+    # Add new verdaccio user:
+    pnpm adduser --registry=https://verdaccio.ubreakit.com
+fi
+
+npmrcPath=~/.npmrc
+envFilePath=.env.local
+
+# Extract token from ~/.npmrc
+token=$(cat ~/.npmrc |  grep -o '\/\/verdaccio\.ubreakit\.com\/:_authToken="[^"]*"' | sed 's/\/\/verdaccio\.ubreakit\.com\/:_authToken="//' | sed 's/"$//')
+
+if [ -z "$token" ]; then
+    echo "Token not found in ~/.npmrc"
+    return 1
+fi
+
+# Update .env.local with VERDACCIO_AUTH_TOKEN
+awk -v token="\"$token\"" '/^VERDACCIO_AUTH_TOKEN=/{$0="VERDACCIO_AUTH_TOKEN=" token}1' .env.local > .env.local.tmp && mv .env.local.tmp .env.local
+
+echo -e "Token successfully copied to .env.local\n\n"
+
+cp backend/.env.example backend/.env
+
+envFilePath="frontend/.env.local"
+exampleEnvFilePath="frontend/.env.local.example"
+algoliaApiKey="\"12345-your-key-here\""
```

Comment:
In general, if you have a field you're supposed to edit in a shell script, I'd recommend putting it at the top since finding a random variable to change in the middle of the file is unintuitive. But that said, since we have interactive questions regarding verdaccio anyway, we might just want to ask interactively (though that could get logged) or use an environment variable. Otherwise this sets up a risk of committing with the change. It might also make switching branches weird (like we do at the top of the file) since we will have a changed file.

Reply:
Pull open to fix this: https://github.com/iFixit/react-commerce/pull/1831